rllib/core/rl_trainer/tf/tf_rl_trainer.py

```diff
@@ -102,10 +101,32 @@ def __init__(
         tf1.enable_eager_execution()
 
         self._enable_tf_function = framework_hyperparameters.eager_tracing
+
+    @override(RLTrainer)
+    def build(self) -> None:
+        # TODO (Kourosh): Does strategy has to be an attribute here? if so it's very
+        # hidden to the user of this class that there is such an attribute.
+        if not self._distributed:
+            if self._use_gpu:
+                # mirrored strategy is typically used for multi-gpu training
+                # on a single machine, however we can use it for single-gpu
+                local_gpu = [f"GPU:{self._local_gpu_id}"]
+                self.strategy = tf.distribute.MirroredStrategy(devices=local_gpu)
+            else:
+                # the default strategy is a no-op that can be used in the local mode
+                # cpu only case
+                self.strategy = tf.distribute.get_strategy()
+        else:
+            self.strategy = tf.distribute.MultiWorkerMirroredStrategy()
+        with self.strategy.scope():
+            super().build()
         if self._enable_tf_function:
-            self._update_fn = tf.function(self._do_update_fn)
+            self._update_fn = tf.function(self._traced_update_helper)
         else:
-            self._update_fn = self._do_update_fn
+            self._update_fn = self._traced_update_helper
+
+    def _traced_update_helper(self, batch):
+        return self.strategy.run(self._do_update_fn, args=(batch,))
```

Comment:
Can you add a comment here, why we don't do self.strategy.run() of a tf.function wrapped update call?